### PR TITLE
Establish tls without negotiating when protocol scheme is amqps

### PR DESCRIPTION
**Pull request by amenzhinsky**
#10

http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#definition-idp861488

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package amqp
```
Comment (vcabbage) [inline client_test.go]: I'd prefer to handle connection setup testing in the integration tests. Please remove this and the other supporting changes.

**Comment by amenzhinsky** [comment]
@vcabbage all done

File: client.go [hunk 2]
```diff
@@ -24,50 +25,59 @@ type Client struct {
...
-// If no port is provided, 5672 will be used.
+// If no port is provided, 5672 will be used for "amqp" and 5671 for "amqps".
 func Dial(addr string, opts ...ConnOption) (*Client, error) {
 	u, err := url.Parse(addr)
 	if err != nil {
 		return nil, err
 	}
+	if u.Scheme == "" {
+		u.Scheme = "amqp"
+	}
```
Comment (vcabbage) [inline client.go]: Can you remove this? It shouldn't be necessary anymore.

Comment (amenzhinsky) [reply client.go]: Forgot to do this in the last commit. Removed.

File: client.go [hunk 3]
```diff
@@ -24,50 +25,55 @@ type Client struct {
...
 
 // New establishes an AMQP client connection over conn.
 func New(conn net.Conn, opts ...ConnOption) (*Client, error) {
 	c, err := newConn(conn, opts...)
-	return &Client{conn: c}, err
+	if err != nil {
+		return nil, err
+	}
+	err = c.start()
+	return &Client{conn: c}, c.start()
```
Comment (vcabbage) [inline client.go]: Looks like an inadvertent extra call to `c.start()` got left in. Should be ready to merge once that's corrected.

Comment (amenzhinsky) [reply client.go]: fixed

**Pull request merged**